sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosContainerChildResourceKind.java

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.models;
+
+/**
+ * Specifies the kind of resource that has a Cosmos container as parent resource.
+ */
+public enum CosmosContainerChildResourceKind {
```

Comment:
We have `ResourceType` and `CosmosResourceType`. 
Should we rename this to `ContainerChildResourceType` ?
@kirankumarkolli  @moderakh  thoughts ?

Reply:
The naming seems a bit odd to me. why can't we use the existing `CosmosResourceType`?

Reply:
CosmosResourceType is "implementation" and it also includes more resource types than what is applicable to creating a Permission object.
I renamed the class however to ContainerChildResourceType.

Reply:
@kirankumarkolli please comment on the naming.